### PR TITLE
Add default const visibility

### PR DESCRIPTION
**Pull request by simonberger**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| License       | MIT

PHP 7.1 introduced const visibility. As Symfony 3.4 is gone (soon) the default visibility should be set everywhere now. A lot of them were not intended for public use, but we cannot change it easy without a BC.
Adding the default visibility is no break for an inheriting class.

While going through all the automatic changes I also replaced string FQCN to use `::class` in several (older) tests. I somehow regretted starting it because they appeared a lot but finished changing all in tests with changed visibility.
I have no problem removing them from this PR again if that's a review burden.

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
This is going to be a nightmare to merge into upper branches.
Either this can be automated or we won't do it imho.
I'd be tempted to reject this change.

**Comment by derrabus** [comment]
Hello Simon and thank you for your PR. I agree with Nicolas that we should only apply code style changes if we can automate them. The 4.4 and 5.1 branches have already diverged significantly. I've had the honor of merging my own CS change up recently and I can tell you, Nicolas is not joking.

My problem with this PR is that you're also applying too many CS changes at once. You're changing the constant visibility, converting FQCN strings to `::class` and somewhere you also migrated a nested `dirname` call. We would have to watch out for too many potential problems when merging this up.

Now back to the constant visibility, the original concern this PR was supposed to address. This is in fact a change we can do with PHP CS Fixer pretty easily. I can't speak for Nicolas, but I personally would be fine with a PR that just contains a CS Fixer rule and only automatic changes done with CS Fixer by only applying this very rule.

**Comment by Nyholm** [comment]
I’ve also tried replacing FQCN strings with constants before. :) 

I’m happy with an updated cs-fixer rule. But I’m 👎 for this pr.

**Comment by simonberger** [comment]
As I wrote I have no problem to remove the FQCN changes. I didn't plan to do it that often. The visibility change is automated. I think there are various ways and tools to do it. Even a regex replace should be enough.
The dir change came from fobbot. I added it in the patch commit.
I'll push the visibility modifier only, later today then you can have a look if you want to merge it or like to do it on your own.
It's really to easy to do to skip it IMO.

**Comment by Nyholm** [comment]
I understand that you have done a regex search or automated it somehow. But it needs to be automated from Symfony perspective too. That is why using php-cs-fixer was suggested.

**Comment by simonberger** [comment]
> But it needs to be automated from Symfony perspective too. That is why using php-cs-fixer was suggested.

If it is done I really don't mind how and by whom 👍

**Comment by simonberger** [comment]
Here are the visibility modifier changes only.
I simulated merging up to 5.1 and got const conflicts in 19 classes (+6 in deleted classes). Not too hefty or?

I like the idea to do it in the php-cs-fixer. Also for future changes. Just close the PR in case you are ready to do so.

**Comment by jderusse** [comment]
What's the benefit of switching from not defined to all public?

I'd rather define real visibility class by class but it's a lot of work and BC break and conflict hell for mergers...

Thus I'm voting 👎 for this change, but rather recommand using vibility on new classes only.

**Comment by simonberger** [comment]
> What's the benefit of switching from not defined to all public?
> 
> I'd rather define real visibility class by class but it's a lot of work and BC break and conflict hell for mergers...
> 
> Thus I'm voting 👎 for this change, but rather recommand using vibility on new classes only.

As you wrote adding the real visibility would be cool but unfortunately it's impossible or not worth to do.
There is no measurable improvement in performance or whatever but that's it with almost all CS changes.
My two cents: Why not invest a clear and relatively short piece of time to have a more consistent codebase and make IDEs happy. For me there is always something missing if I look at such a const. That's enough reason for me to want to change it.

**Comment by Nyholm** [comment]
Simon: We are happy to change it. 

But it needs to be changed as a CS fix. Ie, we need to activate a php-cs-fixer rule that does this change. 

Im closing this PR now as it has a few down votes and the reasons pointed out repeatedly in this thread. I encourage you to open a new PR with this rule change in php-cs-fixer config.

File: src/Symfony/Component/Debug/Tests/DebugClassLoaderTest.php [hunk 1]
```diff
@@ -380,7 +380,7 @@ public function findFile($class)
         if (TestingUnsilencing::class === $class) {
             eval('-- parse error --');
         } elseif (TestingStacking::class === $class) {
-            eval('namespace '.__NAMESPACE__.'; class TestingStacking { function foo() {} }');
+            eval('namespace '.__NAMESPACE__.'; class TestingStacking { publicfunction foo() {} }');
```
Comment (derrabus) [inline src/Symfony/Component/Debug/Tests/DebugClassLoaderTest.php:383]: This does not look right. 👀

**Comment by derrabus** [comment]
Simon, I don't want to discourage you from contributing. And if I recall correctly, this is at least your second attempt to make your first contribution through CS changes.

We really appreciate contributions to the Symfony codebase, but tbh CS changes are not the best way to do this.

As for this particular change, I would feel more comfortable if a core team member did this with the help of PHP CS Fixer because that person could also take care of the merges.

**Comment by simonberger** [comment]
> Simon, I don't want to discourage you from contributing. And if I recall correctly, this is at least your second attempt to make your first contribution through CS changes.
> 
> We really appreciate contributions to the Symfony codebase, but tbh CS changes are not the best way to do this.
> 
> As for this particular change, I would feel more comfortable if a core team member did this with the help of PHP CS Fixer because that person could also take care of the merges.

Thank you for your kind words! Don't worry, it's not my goal in the first place to do a contribution to Symfony more to change things I don't like in a framework I use a lot. I see your point. If a closed PR from me is can be a starting shot for something I am also perfectly fine with that.

**Pull request closed without merging**